File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
N = int(input())
result = 0

while N>0:
    if N%5==0:
        N = int(N/5)
        result += N
        print(result)
        break
    N -= 3
    result += 1
    if N==0:
        print(result)
    elif N<0:
        print(-1)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	
	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int result = 0;

		while (N>0) {
			if (N%5 == 0) {
				N = (int)N/5;
				result += N;
				System.out.print(result);
				break;
			}
			N -= 3;
			result++;
			
			if (N == 0) {
				System.out.print(result);
			} else if (N < 0) {
				System.out.print(-1);
			}
		}
		
		
	}
}